dlt/destinations/impl/lancedb/lancedb_client.py

```diff
@@ -0,0 +1,698 @@
+            self.db_client.drop_table(table_name)
+
+        self._delete_sentinel_table()
+
+    @lancedb_error
+    def initialize_storage(self, truncate_tables: Iterable[str] = None) -> None:
+        if not self.is_storage_initialized():
+            self._create_sentinel_table()
+        elif truncate_tables:
+            for table_name in truncate_tables:
+                fq_table_name = self.make_qualified_table_name(table_name)
+                if not self.table_exists(fq_table_name):
+                    continue
+                schema = self.get_table_schema(fq_table_name)
+                self.db_client.drop_table(fq_table_name)
+                self.create_table(
+                    table_name=fq_table_name,
+                    schema=schema,
+                )
+
+    @lancedb_error
+    def is_storage_initialized(self) -> bool:
+        return self.table_exists(self.sentinel_table)
+
+    def _create_sentinel_table(self) -> Table:
+        """Create an empty table to indicate that the storage is initialized."""
+        return self.create_table(schema=NULL_SCHEMA, table_name=self.sentinel_table)
+
+    def _delete_sentinel_table(self) -> None:
+        """Delete the sentinel table."""
+        self.db_client.drop_table(self.sentinel_table)
+
+    @lancedb_error
+    def update_stored_schema(
+        self,
+        only_tables: Iterable[str] = None,
+        expected_update: TSchemaTables = None,
+    ) -> Optional[TSchemaTables]:
+        super().update_stored_schema(only_tables, expected_update)
+        applied_update: TSchemaTables = {}
+
+        try:
+            schema_info = self.get_stored_schema_by_hash(self.schema.stored_version_hash)
+        except DestinationUndefinedEntity:
+            schema_info = None
+
+        if schema_info is None:
+            logger.info(
+                f"Schema with hash {self.schema.stored_version_hash} "
+                "not found in the storage. upgrading"
+            )
+            self._execute_schema_update(only_tables)
+        else:
+            logger.info(
+                f"Schema with hash {self.schema.stored_version_hash} "
+                f"inserted at {schema_info.inserted_at} found "
+                "in storage, no upgrade required"
+            )
+        return applied_update
+
+    def get_storage_table(self, table_name: str) -> Tuple[bool, TTableSchemaColumns]:
+        table_schema: TTableSchemaColumns = {}
+
+        try:
+            fq_table_name = self.make_qualified_table_name(table_name)
+            arrow_schema = self.db_client.open_table(fq_table_name).schema
+        except FileNotFoundError:
+            return False, table_schema
+        except Exception:
+            raise
+
+        for field_name, field_type in arrow_schema_to_dict(arrow_schema).items():
+            schema_c: TColumnSchema = {
+                "name": self.schema.naming.normalize_identifier(field_name),
+                **self._from_db_type(field_type, None, None),
```

Comment:
Do you not need to send precision data from the arrow schema to the typemapper here? You will loose this info if you don't, I think tests should fail on this too

Reply:
The precision and scale, if present, is obtained from the arrow schema by way of `from_db_type`:

https://github.com/dlt-hub/dlt/blob/9322477461288b77ee317232ee62fcc0897ab969/dlt/destinations/impl/lancedb/lancedb_client.py#L122-L145

To make this clearer to the reader, I've removed the None arguments:

https://github.com/dlt-hub/dlt/blob/9322477461288b77ee317232ee62fcc0897ab969/dlt/destinations/impl/lancedb/lancedb_client.py#L393-L397